### ISSUE TITLE
math.bits : make-bits on 0 returns sequence of length 0

### ISSUE THREAD

**Issue by nomennescio** [factor/factor#2632]
```
0 make-bits
T{ bits f 0 0 }
```
i.e. a vector of length 0, whereas it should be a vector of length 1.
Compare with
```
1 make-bits
T{ bits f 1 1 }
```

**Comment by mrjbq7** [comment]
I have no idea if we should change this or not:

https://math.stackexchange.com/questions/3008663/number-of-bits-in-0

**Comment by mrjbq7** [comment]
If we do, this is the patch.

```diff
diff --git a/basis/math/bits/bits.factor b/basis/math/bits/bits.factor
index 4b94a3c733..9f345986aa 100644
--- a/basis/math/bits/bits.factor
+++ b/basis/math/bits/bits.factor
@@ -11,7 +11,7 @@ C: <bits> bits
 
 : make-bits ( number -- bits )
     check-negative-bits
-    [ T{ bits f 0 0 } ] [ dup abs log2 1 + <bits> ] if-zero ; inline
+    [ T{ bits f 0 1 } ] [ dup abs log2 1 + <bits> ] if-zero ; inline
 
 M: bits length length>> ; inline
```

**Comment by nomennescio** [comment]
> I have no idea if we should change this or not:
> 
> https://math.stackexchange.com/questions/3008663/number-of-bits-in-0

I  think the information theoretic "log" approach is only valid for positive numbers. Zero is a special case. It still requires 1 bit to store the number 0. Check out e,g. Huffman encoding to see how that works out.

If you look at the description of class `bits` it has : 
```Tuple representing a number as a virtual sequence of booleans. The first bit is the least significant bit.```
1 bit can represent exactly two values: `0` and `1`, 2 bits 4 values [0..2^2-1],  n bits 2^n values [0..2^n-1].
So the smallest representation of number 0 is 1 bit with value 0.

Hope this convinces you.

**Comment by mrjbq7** [comment]
I don't quite understand why it was explicitly zero length before, so I'm not sure that making it have one length now is without consequence, but let's give it a try.

Fixed in ab2d1bbb50996d3611bf3a3f8eccb9a33d4119ca.
